packages/piro/test/Main_AnalysisDriver_Tpetra.cpp

```diff
@@ -77,11 +83,8 @@ int main(int argc, char *argv[]) {
   // Initialize MPI
   Teuchos::GlobalMPISession mpiSession(&argc,&argv);
   int Proc=mpiSession.getRank();
-#ifdef HAVE_MPI
-  MPI_Comm appComm = MPI_COMM_WORLD;
-#else
-  int appComm=0;
-#endif
+
+  auto appComm = Tpetra::getDefaultComm();
```

Comment:
Much better; thanks! :D  btw if you use `Tpetra::ScopeGuard`, you can get rid of `Teuchos::GlobalMPISession`, and just use the comm that comes from `Tpetra::getDefaultComm()` to get the rank.  That's for next time, though :-) .